src/client/Microsoft.Identity.Client.Broker/RuntimeBroker.cs

```diff
@@ -227,6 +228,40 @@ private IntPtr GetParentWindow(CoreUIParent uiParent)
             return msalTokenResponse;
         }
 
+        public async Task<MsalTokenResponse> AcquireTokenByUsernamePasswordAsync(
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenByUsernamePasswordParameters acquireTokenByUsernamePasswordParameters)
+        {
+            var cancellationToken = authenticationRequestParameters.RequestContext.UserCancellationToken;
+            MsalTokenResponse msalTokenResponse = null;
+
+            _logger.Verbose("[WamBroker] Acquiring token with Username Password flow.");
+
+            using (var core = new NativeInterop.Core())
+            using (AuthParameters authParams = WamAdapters.GetCommonAuthParameters(authenticationRequestParameters, _wamOptions.MsaPassthrough))
+            {
+                authParams.Properties["MSALRuntime_Username"] = acquireTokenByUsernamePasswordParameters.Username;
+                authParams.Properties["MSALRuntime_Password"] = new System.Net.NetworkCredential(string.Empty, acquireTokenByUsernamePasswordParameters.Password).Password;
```

Comment:
To transform from a SecureString to a string, please use https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/src/client/Microsoft.Identity.Client/Internal/Requests/UsernamePasswordRequest.cs